# Recommendation Algorithm v4
Focus on data engineering and strategies instead of model

## Part 1: Update stock database
Please use `conda env:talib`

In [1]:
import coremlv2 as core
core._init_calc()
import importlib
importlib.reload(core)

<module 'coremlv2' from 'C:\\Users\\jonat\\Documents\\#PROJECT\\idx\\coremlv2.py'>

In [2]:
ROOT_PATH = './'
period = 'max'
sleep = 1.5
abbr = 'id'
core.yfinance_db_v2(abbr, ROOT_PATH=ROOT_PATH, sleep=sleep, period=period)
core.calculate_all_indicators_v2(abbr=abbr, verbose=50, ROOT_PATH=ROOT_PATH)

Elapsed time for 50 requests: 0:00:00.585802 0
Elapsed time for 50 requests: 0:01:26.908419 50
Elapsed time for 50 requests: 0:01:26.534246 100
Elapsed time for 50 requests: 0:01:25.341810 150
Elapsed time for 50 requests: 0:01:25.671791 200
Elapsed time for 50 requests: 0:01:25.440237 250
Elapsed time for 50 requests: 0:01:25.587037 300
Elapsed time for 50 requests: 0:01:25.093327 350
Elapsed time for 50 requests: 0:01:25.228509 400
Elapsed time for 50 requests: 0:01:25.097922 450
Elapsed time for 50 requests: 0:01:25.390231 500
Elapsed time for 50 requests: 0:01:24.293858 550
Elapsed time for 50 requests: 0:01:24.654967 600
Elapsed time for 50 requests: 0:01:24.941568 650
Elapsed time for 50 requests: 0:01:24.725239 700
0/714: BBCA.JK. Calculate total volume.
1/714: BBRI.JK. Calculate total volume.
2/714: TLKM.JK. Calculate total volume.
3/714: BMRI.JK. Calculate total volume.
4/714: ARTO.JK. Calculate total volume.
5/714: ASII.JK. Calculate total volume.
6/714: UNVR.JK. Calculate to

C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

98/714: SMRA.JK. Calculate relative volume.
99/714: MAPI.JK. Calculate relative volume.
100/714: APIC.JK. Calculate relative volume.
101/714: LIFE.JK. Calculate relative volume.
102/714: SMAR.JK. Calculate relative volume.
103/714: MEDC.JK. Calculate relative volume.
104/714: ASSA.JK. Calculate relative volume.
105/714: MCOL.JK. Calculate relative volume.
106/714: CITA.JK. Calculate relative volume.
107/714: TAPG.JK. Calculate relative volume.
108/714: ARCI.JK. Calculate relative volume.
109/714: FAPA.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

110/714: BSSR.JK. Calculate relative volume.
111/714: BJTM.JK. Calculate relative volume.
112/714: MNCN.JK. Calculate relative volume.
113/714: LINK.JK. Calculate relative volume.
114/714: LPPF.JK. Calculate relative volume.
115/714: TINS.JK. Calculate relative volume.
116/714: WIKA.JK. Calculate relative volume.
117/714: GGRP.JK. Calculate relative volume.
118/714: POWR.JK. Calculate relative volume.
119/714: SSMS.JK. Calculate relative volume.
120/714: STTP.JK. Calculate relative volume.
121/714: LPKR.JK. Calculate relative volume.
122/714: MTDL.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

123/714: TOBA.JK. Calculate relative volume.
124/714: SOHO.JK. Calculate relative volume.
125/714: DMAS.JK. Calculate relative volume.
126/714: ERAA.JK. Calculate relative volume.
127/714: CMNP.JK. Calculate relative volume.
128/714: EDGE.JK. Calculate relative volume.
129/714: INDY.JK. Calculate relative volume.
130/714: BNBA.JK. Calculate relative volume.
131/714: MCAS.JK. Calculate relative volume.
132/714: WSKT.JK. Calculate relative volume.
133/714: DMND.JK. Calculate relative volume.
134/714: BMAS.JK. Calculate relative volume.
135/714: TFAS.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

136/714: POLL.JK. Calculate relative volume.
137/714: IBST.JK. Calculate relative volume.
138/714: PRDA.JK. Calculate relative volume.
139/714: BBMD.JK. Calculate relative volume.
140/714: SMSM.JK. Calculate relative volume.
141/714: LSIP.JK. Calculate relative volume.
142/714: ROTI.JK. Calculate relative volume.
143/714: TECH.JK. Calculate relative volume.
144/714: KRAS.JK. Calculate relative volume.
145/714: MPRO.JK. Calculate relative volume.
146/714: FILM.JK. Calculate relative volume.
147/714: HERO.JK. Calculate relative volume.
148/714: ESSA.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

149/714: ADMF.JK. Calculate relative volume.
150/714: MAYA.JK. Calculate relative volume.
151/714: EPMT.JK. Calculate relative volume.
152/714: INPP.JK. Calculate relative volume.
153/714: TURI.JK. Calculate relative volume.
154/714: KPIG.JK. Calculate relative volume.
155/714: CENT.JK. Calculate relative volume.
156/714: SIMP.JK. Calculate relative volume.
157/714: MLPT.JK. Calculate relative volume.
158/714: PLIN.JK. Calculate relative volume.
159/714: TMAS.JK. Calculate relative volume.
160/714: BMHS.JK. Calculate relative volume.
161/714: JRPT.JK. Calculate relative volume.
162/714: INAF.JK. Calculate relative volume.
163/714: MAPA.JK. Calculate relative volume.
164/714: TSPC.JK. Calculate relative volume.
165/714: PTPP.JK. Calculate relative volume.
166/714: MIDI.JK. Calculate relative volume.
167/714: TGKA.JK. Calculate relative volume.
168/714: SAME.JK. Calculate relative volume.
169/714: DUTI.JK. Calculate relative volume.
170/714: ARNA.JK. Calculate relative volume.
171/714: S

C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

176/714: SAMF.JK. Calculate relative volume.
177/714: PALM.JK. Calculate relative volume.
178/714: PNLF.JK. Calculate relative volume.
179/714: AUTO.JK. Calculate relative volume.
180/714: UNIC.JK. Calculate relative volume.
181/714: CLEO.JK. Calculate relative volume.
182/714: BOGA.JK. Calculate relative volume.
183/714: DSNG.JK. Calculate relative volume.
184/714: IPTV.JK. Calculate relative volume.
185/714: BUMI.JK. Calculate relative volume.
186/714: BHAT.JK. Calculate relative volume.
187/714: IFSH.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

188/714: BRAM.JK. Calculate relative volume.
189/714: WOOD.JK. Calculate relative volume.
190/714: BABP.JK. Calculate relative volume.
191/714: MPMX.JK. Calculate relative volume.
192/714: BHIT.JK. Calculate relative volume.
193/714: ABDA.JK. Calculate relative volume.
194/714: MARK.JK. Calculate relative volume.
195/714: MLPL.JK. Calculate relative volume.
196/714: SDRA.JK. Calculate relative volume.
197/714: PSGO.JK. Calculate relative volume.
198/714: RALS.JK. Calculate relative volume.
199/714: AGII.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

200/714: AMOR.JK. Calculate relative volume.
201/714: MBAP.JK. Calculate relative volume.
202/714: MASB.JK. Calculate relative volume.
203/714: WMPP.JK. Calculate relative volume.
204/714: MSKY.JK. Calculate relative volume.
205/714: MCOR.JK. Calculate relative volume.
206/714: DNAR.JK. Calculate relative volume.
207/714: TBLA.JK. Calculate relative volume.
208/714: BMTR.JK. Calculate relative volume.
209/714: UFOE.JK. Calculate relative volume.
210/714: RISE.JK. Calculate relative volume.
211/714: MMLP.JK. Calculate relative volume.
212/714: FAST.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

213/714: ABMM.JK. Calculate relative volume.
214/714: HEXA.JK. Calculate relative volume.
215/714: BKSW.JK. Calculate relative volume.
216/714: SRAJ.JK. Calculate relative volume.
217/714: MYOH.JK. Calculate relative volume.
218/714: SGRO.JK. Calculate relative volume.
219/714: BPII.JK. Calculate relative volume.
220/714: AMAR.JK. Calculate relative volume.
221/714: BKSL.JK. Calculate relative volume.
222/714: PPRO.JK. Calculate relative volume.
223/714: CMPP.JK. Calculate relative volume.
224/714: IMJS.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

225/714: BOLA.JK. Calculate relative volume.
226/714: BUKK.JK. Calculate relative volume.
227/714: MAPB.JK. Calculate relative volume.
228/714: IMAS.JK. Calculate relative volume.
229/714: SMDR.JK. Calculate relative volume.
230/714: VICI.JK. Calculate relative volume.
231/714: BIRD.JK. Calculate relative volume.
232/714: PSAB.JK. Calculate relative volume.
233/714: DRMA.JK. Calculate relative volume.
234/714: NIRO.JK. Calculate relative volume.
235/714: ADHI.JK. Calculate relative volume.
236/714: KIJA.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

237/714: BALI.JK. Calculate relative volume.
238/714: FISH.JK. Calculate relative volume.
239/714: ANJT.JK. Calculate relative volume.
240/714: IRRA.JK. Calculate relative volume.
241/714: DEPO.JK. Calculate relative volume.
242/714: YULE.JK. Calculate relative volume.
243/714: LPCK.JK. Calculate relative volume.
244/714: PEGE.JK. Calculate relative volume.
245/714: TFCO.JK. Calculate relative volume.
246/714: PNBS.JK. Calculate relative volume.
247/714: ASRI.JK. Calculate relative volume.
248/714: NOBU.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

249/714: KINO.JK. Calculate relative volume.
250/714: DVLA.JK. Calculate relative volume.
251/714: BULL.JK. Calculate relative volume.
252/714: MPPA.JK. Calculate relative volume.
253/714: PBID.JK. Calculate relative volume.
254/714: BESS.JK. Calculate relative volume.
255/714: BISI.JK. Calculate relative volume.
256/714: MGRO.JK. Calculate relative volume.
257/714: DLTA.JK. Calculate relative volume.
258/714: CSMI.JK. Calculate relative volume.
259/714: ISSP.JK. Calculate relative volume.
260/714: TUGU.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

261/714: MFIN.JK. Calculate relative volume.
262/714: RANC.JK. Calculate relative volume.
263/714: POLI.JK. Calculate relative volume.
264/714: MTLA.JK. Calculate relative volume.
265/714: AGRS.JK. Calculate relative volume.
266/714: WSBP.JK. Calculate relative volume.
267/714: KMTR.JK. Calculate relative volume.
268/714: PNIN.JK. Calculate relative volume.
269/714: SURE.JK. Calculate relative volume.
270/714: APLN.JK. Calculate relative volume.
271/714: INDR.JK. Calculate relative volume.
272/714: BLTZ.JK. Calculate relative volume.
273/714: MLIA.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

274/714: ACST.JK. Calculate relative volume.
275/714: HITS.JK. Calculate relative volume.
276/714: ZINC.JK. Calculate relative volume.
277/714: BGTG.JK. Calculate relative volume.
278/714: SHIP.JK. Calculate relative volume.
279/714: INPC.JK. Calculate relative volume.
280/714: TRIM.JK. Calculate relative volume.
281/714: FPNI.JK. Calculate relative volume.
282/714: SIPD.JK. Calculate relative volume.
283/714: BPFI.JK. Calculate relative volume.
284/714: NIKL.JK. Calculate relative volume.
285/714: GJTL.JK. Calculate relative volume.
286/714: BWPT.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

287/714: BCAP.JK. Calculate relative volume.
288/714: BTEK.JK. Calculate relative volume.
289/714: ASMI.JK. Calculate relative volume.
290/714: MARI.JK. Calculate relative volume.
291/714: CSAP.JK. Calculate relative volume.
292/714: MREI.JK. Calculate relative volume.
293/714: BIPI.JK. Calculate relative volume.
294/714: SPMA.JK. Calculate relative volume.
295/714: PTRO.JK. Calculate relative volume.
296/714: TIFA.JK. Calculate relative volume.
297/714: TOTO.JK. Calculate relative volume.
298/714: WTON.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

299/714: ELTY.JK. Calculate relative volume.
300/714: PSSI.JK. Calculate relative volume.
301/714: SCCO.JK. Calculate relative volume.
302/714: DOID.JK. Calculate relative volume.
303/714: CLAY.JK. Calculate relative volume.
304/714: ADES.JK. Calculate relative volume.
305/714: PORT.JK. Calculate relative volume.
306/714: SSIA.JK. Calculate relative volume.
307/714: SFAN.JK. Calculate relative volume.
308/714: AMAN.JK. Calculate relative volume.
309/714: BVIC.JK. Calculate relative volume.
310/714: AMFG.JK. Calculate relative volume.
311/714: ELSA.JK. Calculate relative volume.
312/714: META.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

313/714: GSMF.JK. Calculate relative volume.
314/714: PZZA.JK. Calculate relative volume.
315/714: CTBN.JK. Calculate relative volume.
316/714: VICO.JK. Calculate relative volume.
317/714: GMFI.JK. Calculate relative volume.
318/714: PGUN.JK. Calculate relative volume.
319/714: MDIA.JK. Calculate relative volume.
320/714: WMUU.JK. Calculate relative volume.
321/714: JKON.JK. Calculate relative volume.
322/714: PMJS.JK. Calculate relative volume.
323/714: RDTX.JK. Calculate relative volume.
324/714: MBSS.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

325/714: TAMU.JK. Calculate relative volume.
326/714: SGER.JK. Calculate relative volume.
327/714: WEGE.JK. Calculate relative volume.
328/714: SMKL.JK. Calculate relative volume.
329/714: APEX.JK. Calculate relative volume.
330/714: URBN.JK. Calculate relative volume.
331/714: JTPE.JK. Calculate relative volume.
332/714: PPRE.JK. Calculate relative volume.
333/714: SLIS.JK. Calculate relative volume.
334/714: BACA.JK. Calculate relative volume.
335/714: KEJU.JK. Calculate relative volume.
336/714: BOLT.JK. Calculate relative volume.
337/714: MERK.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

338/714: JSPT.JK. Calculate relative volume.
339/714: DWGL.JK. Calculate relative volume.
340/714: SPTO.JK. Calculate relative volume.
341/714: CAMP.JK. Calculate relative volume.
342/714: HOKI.JK. Calculate relative volume.
343/714: AMAG.JK. Calculate relative volume.
344/714: TOPS.JK. Calculate relative volume.
345/714: SONA.JK. Calculate relative volume.
346/714: KEEN.JK. Calculate relative volume.
347/714: KREN.JK. Calculate relative volume.
348/714: TRST.JK. Calculate relative volume.
349/714: MTFN.JK. Calculate relative volume.
350/714: RSGK.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

351/714: DILD.JK. Calculate relative volume.
352/714: IPCM.JK. Calculate relative volume.
353/714: ATIC.JK. Calculate relative volume.
354/714: PTSP.JK. Calculate relative volume.
355/714: MAIN.JK. Calculate relative volume.
356/714: ALDO.JK. Calculate relative volume.
357/714: OMRE.JK. Calculate relative volume.
358/714: SKLT.JK. Calculate relative volume.
359/714: IFII.JK. Calculate relative volume.
360/714: BCIC.JK. Calculate relative volume.
361/714: PRIM.JK. Calculate relative volume.
362/714: INDS.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

363/714: SOCI.JK. Calculate relative volume.
364/714: KKGI.JK. Calculate relative volume.
365/714: SKRN.JK. Calculate relative volume.
366/714: GWSA.JK. Calculate relative volume.
367/714: DIVA.JK. Calculate relative volume.
368/714: BLTA.JK. Calculate relative volume.
369/714: GHON.JK. Calculate relative volume.
370/714: HOMI.JK. Calculate relative volume.
371/714: PTSN.JK. Calculate relative volume.
372/714: CSRA.JK. Calculate relative volume.
373/714: HDFA.JK. Calculate relative volume.
374/714: PNGO.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

375/714: MOLI.JK. Calculate relative volume.
376/714: RODA.JK. Calculate relative volume.
377/714: ALMI.JK. Calculate relative volume.
378/714: INPS.JK. Calculate relative volume.
379/714: BRNA.JK. Calculate relative volume.
380/714: ASGR.JK. Calculate relative volume.
381/714: KBLI.JK. Calculate relative volume.
382/714: JGLE.JK. Calculate relative volume.
383/714: DGIK.JK. Calculate relative volume.
384/714: CEKA.JK. Calculate relative volume.
385/714: IPOL.JK. Calculate relative volume.
386/714: LTLS.JK. Calculate relative volume.
387/714: PMMP.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

388/714: ROCK.JK. Calculate relative volume.
389/714: TRIN.JK. Calculate relative volume.
390/714: PBSA.JK. Calculate relative volume.
391/714: DEWA.JK. Calculate relative volume.
392/714: TCID.JK. Calculate relative volume.
393/714: BEST.JK. Calculate relative volume.
394/714: TOTL.JK. Calculate relative volume.
395/714: ECII.JK. Calculate relative volume.
396/714: BNBR.JK. Calculate relative volume.
397/714: ENRG.JK. Calculate relative volume.
398/714: SAPX.JK. Calculate relative volume.
399/714: WIFI.JK. Calculate relative volume.
400/714: FMII.JK. Calculate relative volume.
401/714: JAWA.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

402/714: BELL.JK. Calculate relative volume.
403/714: KBLV.JK. Calculate relative volume.
404/714: NELY.JK. Calculate relative volume.
405/714: EKAD.JK. Calculate relative volume.
406/714: DEFI.JK. Calculate relative volume.
407/714: RMKE.JK. Calculate relative volume.
408/714: CFIN.JK. Calculate relative volume.
409/714: TPMA.JK. Calculate relative volume.
410/714: HRTA.JK. Calculate relative volume.
411/714: PBRX.JK. Calculate relative volume.
412/714: IPCC.JK. Calculate relative volume.
413/714: CCSI.JK. Calculate relative volume.
414/714: SSTM.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

415/714: INRU.JK. Calculate relative volume.
416/714: GTSI.JK. Calculate relative volume.
417/714: UVCR.JK. Calculate relative volume.
418/714: UANG.JK. Calculate relative volume.
419/714: CASS.JK. Calculate relative volume.
420/714: MDLN.JK. Calculate relative volume.
421/714: DGNS.JK. Calculate relative volume.
422/714: PEHA.JK. Calculate relative volume.
423/714: WIIM.JK. Calculate relative volume.
424/714: PSKT.JK. Calculate relative volume.
425/714: CITY.JK. Calculate relative volume.
426/714: SMDM.JK. Calculate relative volume.
427/714: IPPE.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

428/714: JIHD.JK. Calculate relative volume.
429/714: PANS.JK. Calculate relative volume.
430/714: TNCA.JK. Calculate relative volume.
431/714: PJAA.JK. Calculate relative volume.
432/714: SHID.JK. Calculate relative volume.
433/714: ARTA.JK. Calculate relative volume.
434/714: JECC.JK. Calculate relative volume.
435/714: ABBA.JK. Calculate relative volume.
436/714: VRNA.JK. Calculate relative volume.
437/714: WINS.JK. Calculate relative volume.
438/714: WOMF.JK. Calculate relative volume.
439/714: GDST.JK. Calculate relative volume.
440/714: RELI.JK. Calculate relative volume.
441/714: VIVA.JK. Calculate relative volume.
442/714: TEBE.JK. Calculate relative volume.
443/714: FUJI.JK. Calculate relative volume.
444/714: RAJA.JK. Calculate relative volume.
445/714: TRUE.JK. Calculate relative volume.
446/714: KOTA.JK. Calculate relative volume.
447/714: LPGI.JK. Calculate relative volume.
448/714: SQMI.JK. Calculate relative volume.
449/714: ARII.JK. Calculate relative volume.
450/714: A

C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

453/714: CARS.JK. Calculate relative volume.
454/714: KIAS.JK. Calculate relative volume.
455/714: PANI.JK. Calculate relative volume.
456/714: VOKS.JK. Calculate relative volume.
457/714: STAR.JK. Calculate relative volume.
458/714: MFMI.JK. Calculate relative volume.
459/714: DAYA.JK. Calculate relative volume.
460/714: AISA.JK. Calculate relative volume.
461/714: IATA.JK. Calculate relative volume.
462/714: DFAM.JK. Calculate relative volume.
463/714: NRCA.JK. Calculate relative volume.
464/714: AKPI.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

465/714: NICL.JK. Calculate relative volume.
466/714: ITMA.JK. Calculate relative volume.
467/714: MYTX.JK. Calculate relative volume.
468/714: IKAI.JK. Calculate relative volume.
469/714: TRIS.JK. Calculate relative volume.
470/714: DKFT.JK. Calculate relative volume.
471/714: TRJA.JK. Calculate relative volume.
472/714: SMMT.JK. Calculate relative volume.
473/714: BATA.JK. Calculate relative volume.
474/714: NASA.JK. Calculate relative volume.
475/714: FIRE.JK. Calculate relative volume.
476/714: TBMS.JK. Calculate relative volume.
477/714: PTPW.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

478/714: BBLD.JK. Calculate relative volume.
479/714: BPTR.JK. Calculate relative volume.
480/714: BKDP.JK. Calculate relative volume.
481/714: SKBM.JK. Calculate relative volume.
482/714: BAJA.JK. Calculate relative volume.
483/714: TARA.JK. Calculate relative volume.
484/714: ARGO.JK. Calculate relative volume.
485/714: PADI.JK. Calculate relative volume.
486/714: YPAS.JK. Calculate relative volume.
487/714: GEMA.JK. Calculate relative volume.
488/714: EMDE.JK. Calculate relative volume.
489/714: ALTO.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

490/714: GDYR.JK. Calculate relative volume.
491/714: KOBX.JK. Calculate relative volume.
492/714: PYFA.JK. Calculate relative volume.
493/714: MITI.JK. Calculate relative volume.
494/714: REAL.JK. Calculate relative volume.
495/714: PNSE.JK. Calculate relative volume.
496/714: WICO.JK. Calculate relative volume.
497/714: SCNP.JK. Calculate relative volume.
498/714: MDKI.JK. Calculate relative volume.
499/714: BSML.JK. Calculate relative volume.
500/714: ASRM.JK. Calculate relative volume.
501/714: TAXI.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

502/714: GAMA.JK. Calculate relative volume.
503/714: INOV.JK. Calculate relative volume.
504/714: SWAT.JK. Calculate relative volume.
505/714: HDIT.JK. Calculate relative volume.
506/714: POLA.JK. Calculate relative volume.
507/714: GOLD.JK. Calculate relative volume.
508/714: NICK.JK. Calculate relative volume.
509/714: ANDI.JK. Calculate relative volume.
510/714: KJEN.JK. Calculate relative volume.
511/714: AKSI.JK. Calculate relative volume.
512/714: MAGP.JK. Calculate relative volume.
513/714: CNKO.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

514/714: HAIS.JK. Calculate relative volume.
515/714: KDSI.JK. Calculate relative volume.
516/714: IGAR.JK. Calculate relative volume.
517/714: INDO.JK. Calculate relative volume.
518/714: KOPI.JK. Calculate relative volume.
519/714: GZCO.JK. Calculate relative volume.
520/714: GLVA.JK. Calculate relative volume.
521/714: BAYU.JK. Calculate relative volume.
522/714: PURA.JK. Calculate relative volume.
523/714: TALF.JK. Calculate relative volume.
524/714: MGLV.JK. Calculate relative volume.
525/714: IDPR.JK. Calculate relative volume.
526/714: KBAG.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

527/714: EAST.JK. Calculate relative volume.
528/714: GPRA.JK. Calculate relative volume.
529/714: BEKS.JK. Calculate relative volume.
530/714: PURI.JK. Calculate relative volume.
531/714: AGAR.JK. Calculate relative volume.
532/714: LPIN.JK. Calculate relative volume.
533/714: DADA.JK. Calculate relative volume.
534/714: ZONE.JK. Calculate relative volume.
535/714: FITT.JK. Calculate relative volume.
536/714: CASH.JK. Calculate relative volume.
537/714: KIOS.JK. Calculate relative volume.
538/714: PCAR.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

539/714: KMDS.JK. Calculate relative volume.
540/714: POLU.JK. Calculate relative volume.
541/714: ZBRA.JK. Calculate relative volume.
542/714: NZIA.JK. Calculate relative volume.
543/714: TOYS.JK. Calculate relative volume.
544/714: DYAN.JK. Calculate relative volume.
545/714: MINA.JK. Calculate relative volume.
546/714: SRSN.JK. Calculate relative volume.
547/714: SBMA.JK. Calculate relative volume.
548/714: AKKU.JK. Calculate relative volume.
549/714: HRME.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

550/714: TRUS.JK. Calculate relative volume.
551/714: LCKM.JK. Calculate relative volume.
552/714: LPLI.JK. Calculate relative volume.
553/714: CLPI.JK. Calculate relative volume.
554/714: BINO.JK. Calculate relative volume.
555/714: KONI.JK. Calculate relative volume.
556/714: PAMG.JK. Calculate relative volume.
557/714: BTON.JK. Calculate relative volume.
558/714: PANR.JK. Calculate relative volume.
559/714: IKBI.JK. Calculate relative volume.
560/714: NPGF.JK. Calculate relative volume.
561/714: BMSR.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

562/714: SNLK.JK. Calculate relative volume.
563/714: NASI.JK. Calculate relative volume.
564/714: OASA.JK. Calculate relative volume.
565/714: UNSP.JK. Calculate relative volume.
566/714: LPPS.JK. Calculate relative volume.
567/714: INCF.JK. Calculate relative volume.
568/714: AYLS.JK. Calculate relative volume.
569/714: BAPI.JK. Calculate relative volume.
570/714: UNIQ.JK. Calculate relative volume.
571/714: ERTX.JK. Calculate relative volume.
572/714: RUNS.JK. Calculate relative volume.
573/714: SOTS.JK. Calculate relative volume.
574/714: MKNT.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

575/714: APLI.JK. Calculate relative volume.
576/714: BIPP.JK. Calculate relative volume.
577/714: HELI.JK. Calculate relative volume.
578/714: BOBA.JK. Calculate relative volume.
579/714: FLMC.JK. Calculate relative volume.
580/714: PTDU.JK. Calculate relative volume.
581/714: CINT.JK. Calculate relative volume.
582/714: COCO.JK. Calculate relative volume.
583/714: ITIC.JK. Calculate relative volume.
584/714: TIRA.JK. Calculate relative volume.
585/714: TGRA.JK. Calculate relative volume.
586/714: SOSS.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

587/714: VINS.JK. Calculate relative volume.
588/714: KBLM.JK. Calculate relative volume.
589/714: MDRN.JK. Calculate relative volume.
590/714: YELO.JK. Calculate relative volume.
591/714: LUCK.JK. Calculate relative volume.
592/714: PSDN.JK. Calculate relative volume.
593/714: BBSS.JK. Calculate relative volume.
594/714: MICE.JK. Calculate relative volume.
595/714: SBAT.JK. Calculate relative volume.
596/714: LAND.JK. Calculate relative volume.
597/714: ALKA.JK. Calculate relative volume.
598/714: GLOB.JK. Calculate relative volume.
599/714: MTWI.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

600/714: ESTI.JK. Calculate relative volume.
601/714: INTA.JK. Calculate relative volume.
602/714: TAYS.JK. Calculate relative volume.
603/714: LEAD.JK. Calculate relative volume.
604/714: RBMS.JK. Calculate relative volume.
605/714: MTPS.JK. Calculate relative volume.
606/714: PDES.JK. Calculate relative volume.
607/714: AHAP.JK. Calculate relative volume.
608/714: APII.JK. Calculate relative volume.
609/714: POLY.JK. Calculate relative volume.
610/714: JSKY.JK. Calculate relative volume.
611/714: AMIN.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

612/714: PGLI.JK. Calculate relative volume.
613/714: ATAP.JK. Calculate relative volume.
614/714: PTIS.JK. Calculate relative volume.
615/714: LMPI.JK. Calculate relative volume.
616/714: WAPO.JK. Calculate relative volume.
617/714: RIGS.JK. Calculate relative volume.
618/714: SULI.JK. Calculate relative volume.
619/714: MIRA.JK. Calculate relative volume.
620/714: EPAC.JK. Calculate relative volume.
621/714: ASDM.JK. Calculate relative volume.
622/714: INAI.JK. Calculate relative volume.
623/714: DSFI.JK. Calculate relative volume.
624/714: PRAS.JK. Calculate relative volume.
625/714: BBRM.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

626/714: HOTL.JK. Calculate relative volume.
627/714: CSIS.JK. Calculate relative volume.
628/714: SDPC.JK. Calculate relative volume.
629/714: LION.JK. Calculate relative volume.
630/714: HOPE.JK. Calculate relative volume.
631/714: HKMU.JK. Calculate relative volume.
632/714: JMAS.JK. Calculate relative volume.
633/714: DIGI.JK. Calculate relative volume.
634/714: WEHA.JK. Calculate relative volume.
635/714: CBMF.JK. Calculate relative volume.
636/714: BIMA.JK. Calculate relative volume.
637/714: BLUE.JK. Calculate relative volume.
638/714: RUIS.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

639/714: WGSH.JK. Calculate relative volume.
640/714: BIKA.JK. Calculate relative volume.
641/714: ESTA.JK. Calculate relative volume.
642/714: JAST.JK. Calculate relative volume.
643/714: KOIN.JK. Calculate relative volume.
644/714: LABA.JK. Calculate relative volume.
645/714: TMPO.JK. Calculate relative volume.
646/714: INTD.JK. Calculate relative volume.
647/714: JAYA.JK. Calculate relative volume.
648/714: MBTO.JK. Calculate relative volume.
649/714: OKAS.JK. Calculate relative volume.
650/714: OPMS.JK. Calculate relative volume.
651/714: BEEF.JK. Calculate relative volume.
652/714: SATU.JK. Calculate relative volume.
653/714: SDMU.JK. Calculate relative volume.
654/714: SAFE.JK. Calculate relative volume.
655/714: DPNS.JK. Calculate relative volume.
656/714: BCIP.JK. Calculate relative volume.
657/714: LFLO.JK. Calculate relative volume.
658/714: WOWS.JK. Calculate relative volume.
659/714: CANI.JK. Calculate relative volume.
660/714: IDEA.JK. Calculate relative volume.
661/714: S

C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

664/714: IPAC.JK. Calculate relative volume.
665/714: KUAS.JK. Calculate relative volume.
666/714: INCI.JK. Calculate relative volume.
667/714: MRAT.JK. Calculate relative volume.
668/714: PURE.JK. Calculate relative volume.
669/714: ICON.JK. Calculate relative volume.
670/714: ASJT.JK. Calculate relative volume.
671/714: ENZO.JK. Calculate relative volume.
672/714: PUDP.JK. Calculate relative volume.
673/714: HADE.JK. Calculate relative volume.
674/714: PPGL.JK. Calculate relative volume.
675/714: ASBI.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

676/714: ARKA.JK. Calculate relative volume.
677/714: PKPK.JK. Calculate relative volume.
678/714: FORU.JK. Calculate relative volume.
679/714: BOSS.JK. Calculate relative volume.
680/714: AIMS.JK. Calculate relative volume.
681/714: OBMD.JK. Calculate relative volume.
682/714: TRUK.JK. Calculate relative volume.
683/714: GPSO.JK. Calculate relative volume.
684/714: KARW.JK. Calculate relative volume.
685/714: FOOD.JK. Calculate relative volume.
686/714: MPOW.JK. Calculate relative volume.
687/714: IBFN.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

688/714: ESIP.JK. Calculate relative volume.
689/714: KICI.JK. Calculate relative volume.
690/714: IKAN.JK. Calculate relative volume.
691/714: LMAS.JK. Calculate relative volume.
692/714: INDX.JK. Calculate relative volume.
693/714: SOFA.JK. Calculate relative volume.
694/714: LMSH.JK. Calculate relative volume.
695/714: LRNA.JK. Calculate relative volume.
696/714: RICY.JK. Calculate relative volume.
697/714: MTSM.JK. Calculate relative volume.
698/714: CTTH.JK. Calculate relative volume.
699/714: PGJO.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

700/714: LUCY.JK. Calculate relative volume.
701/714: CNTX.JK. Calculate relative volume.
702/714: DEAL.JK. Calculate relative volume.
703/714: PICO.JK. Calculate relative volume.
704/714: TAMA.JK. Calculate relative volume.
705/714: ASPI.JK. Calculate relative volume.
706/714: BAPA.JK. Calculate relative volume.
707/714: KAYU.JK. Calculate relative volume.
708/714: PLAN.JK. Calculate relative volume.
709/714: FIMP.JK. Calculate relative volume.
710/714: DCII.JK. Calculate relative volume.
711/714: OILS.JK. Calculate relative volume.


C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df[f'{ticker}_'] = df[ticker] / df['IDX']
C:\Users\jonat\Documents\#PROJECT\idx\coremlv2.py:598: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.

712/714: ADMR.JK. Calculate relative volume.
713/714: ZYRX.JK. Calculate relative volume.
Current progress: 0/714
Current progress: 50/714
Current progress: 100/714
Current progress: 150/714
Current progress: 200/714
Current progress: 250/714
Current progress: 300/714
Current progress: 350/714
Current progress: 400/714
Current progress: 450/714
Current progress: 500/714
Current progress: 550/714
Current progress: 600/714
Current progress: 650/714
Current progress: 700/714


## Part 2: Model engineering and recommendation
please use `conda env:tensorflow`

In [1]:
import coremlv2 as core
core._init_ml()
import importlib
importlib.reload(core)

core.os.environ['CUDA_VISIBLE_DEVICES'] = '-1'

##### Refer to 'model_progression.ipynb' to find best model metadata to load

In [2]:
# Initial params
split = 0.8
recurrent = 120


# Load model
model_source = '327-3_cl-0.5_idc-1_mid-0.5_ep-15_bz-2048_ldtv-2_sbuffs-1_gen-1_1-7'
model_source = '327-9_cl-0.005_idc-1_mid-0.9_ep-40_bz-2048_ldtv-3_sbuffs-1_gen-1_28-0' # 0.6459 / 0.6173
model_source = '327-9_cl-0.1_idc-1_mid-0.9_ep-40_bz-2048_ldtv-3_sbuffs-1_gen-1_7-5' # 0.6371 / 0.6281

ROOT_PATH = './'

model_source_path = core.os.path.join(ROOT_PATH, f'models/preloaded/{model_source}')
model = core.tf.keras.models.load_model(f'{model_source_path}/model')

In [3]:
# Define selected tickers to be analyzed
ticker_groups = ['LSIP','BWPT','SKBM','BUKK','ACES','VIVA','TCPI','INDF','ICBP','TLKM','CMNP']
# Ticker group 327_9
ticker_groups = ['WEGE','ACES','MYOR','AMRT','AKRA','MEDC','AALI','PRDA','GJTL']
ticker_groups_db = [i + '.JK' for i in ticker_groups]
ticker_groups_expanded = core.np.expand_dims(core.np.array(ticker_groups), axis=1)
# ticker_groups_db_expanded = core.np.expand_dims(core.np.array(ticker_groups_db), axis=1)

# Fetch data version that used to train the model
benchmark_db = 'E:\\#PROJECT\\idx\\db\\v8\\dataset\\vid\\AALI.JK_shift0_interval5_recurrent120_split0.8.hdf5'
with core.h5py.File(benchmark_db, 'r') as f:
    data_version = f.attrs['data_version']

# Fetch standard scaler 
standard_scaler = []
for i, ticker_group_expanded in enumerate(ticker_groups_expanded):
    train_mean, train_std = core.standard_scaler_one_shot(ticker_group_expanded, data_version, ROOT_PATH=ROOT_PATH, split=split)
    standard_scaler.append([train_mean, train_std])

In [4]:
# before: 0,1,2,3 if the db updated after the market closed
shifts_from_last = [0,1,2,3]

# Make predictions
db_path = './indicators_db/id_indicators.db'
db_conn = core.sqlite3.connect(db_path)

prediction_results = []
for shift_from_last in shifts_from_last:
    prediction_result = []
    for j, sticker in enumerate(ticker_groups_db):
        tomorrow_input = core.make_one_shot_input(sticker, db_conn, recurrent, shift_from_last=shift_from_last)
        tomorrow_input_n = (tomorrow_input - standard_scaler[j][0]) / standard_scaler[j][1]
        tomorrow_input_e = core.np.expand_dims(tomorrow_input_n, 0)
        tomorrow_prediction = model.predict(tomorrow_input_e)
        # val_acc = histories[i].history['val_accuracy'][-1]
        # print(sticker, f'val_acc:{val_acc*100:.0f}%',tomorrow_prediction)

        # Convert prediction to text
        tomorrow_prediction_argmax = core.np.argmax(tomorrow_prediction)
        prediction = 'UP' if tomorrow_prediction_argmax == 1 else 'DOWN'
        # print(sticker, prediction)
        prediction_result.append(prediction)
    prediction_results.append(prediction_result)

# Check double down and double up
up = []
down = []
for i, results in enumerate(prediction_results[0]):
    #  == prediction_results[2][i]
    if prediction_results[0][i] == prediction_results[1][i]:
        if results == 'UP':
            up.append(ticker_groups[i])
        elif results == 'DOWN':
            down.append(ticker_groups[i])
print(f'up: {up}')
print(f'down: {down}')

up: []
down: ['WEGE', 'ACES', 'MYOR', 'AMRT', 'AKRA', 'MEDC', 'AALI', 'PRDA', 'GJTL']


ticker_groups = ['WEGE','ACES','MYOR','AMRT','AKRA','MEDC','AALI','PRDA','GJTL']

In [5]:
prediction_results

[['DOWN', 'DOWN', 'DOWN', 'DOWN', 'UP', 'UP', 'DOWN', 'DOWN', 'UP'],
 ['UP', 'DOWN', 'DOWN', 'DOWN', 'UP', 'UP', 'UP', 'DOWN', 'UP'],
 ['DOWN', 'UP', 'DOWN', 'DOWN', 'UP', 'UP', 'UP', 'UP', 'UP'],
 ['DOWN', 'DOWN', 'DOWN', 'DOWN', 'UP', 'UP', 'UP', 'DOWN', 'UP']]

**26 Mei 2022**
TCPI 1010k
TLKM 419k -> 500rb ya
ICBP 825k

**2 Juni 2022** (profit min 1%)
MYOR 170k 2 lot (entry 1735, tp 1850)
AKRA 110k 4 lot (entry 1090, tp 1185)
GJTL 66k sisanya (entry 650, tp 700)

**6 Juni 2022**
0.005 & 0.1
ACES 4 lot (entry berapapun, tp 1075)
AKRA skip

**8 Juni 2022**
ACES 3 up 1 down
GJTL 3 up 1 down

up
AKRA
MEDC
AALI
GJTL

down
WEGE
MYOR
AMRT

**10 Juni 2022**
AKRA 1.2jt
MEDC 63k entry bebas, tp 670, cl rsi <50 50%
GJTL 72k entry bebas, tp 755, cl cross sma rsi 50%

ACES Jual, masuk gjtl

**13 Juni 2022**
AKRA
GJTL
sell MEDC
